File: 4kyu/factorial_tail.js
function zeroes(base, num){
  let ans = num;
  for (let i = 2, j = base; i <= base; i++) {
    if (j % i == 0) {
      let p = 0;
      while (j % i == 0) {
        p++;
        j = ~~(j/i);
      }
      let c = 0;
      let k = num;
      while (k/i > 0) {
        k = ~~(k/i);
        c += k;
      }
      ans = Math.min(ans, ~~(c/p));
    }
  }
  return ans;
}

File: 4kyu/factorial_tail.c
int zeroes(base, num){
  int ans = num;
  for (int i = 2, j = base; i <= base; i++) {
    if (j % i == 0) {
      int p = 0;
      while (j % i == 0) {
        p++;
        j /= i;
      }
      int c = 0;
      int k = num;
      while (k/i > 0) {
        k /= i;
        c += k;
      }
      int tmp = c/p;
      if (ans > tmp) ans = tmp;
    }
  }
  return ans;
}
